crates/biome_js_analyze/src/analyzers/nursery/no_implicit_any_let.rs

```diff
@@ -0,0 +1,90 @@
+use biome_analyze::{context::RuleContext, declare_rule, Ast, Rule, RuleDiagnostic};
+use biome_console::markup;
+use biome_js_syntax::{JsFileSource, JsVariableDeclaration, JsVariableDeclarator};
+
+declare_rule! {
+    /// Restrict use of implicit any type in Typescript.
+    ///
+    /// Typescript variable declaration without any `type` or `initialization` can cause issue later in the code.
+    ///
+    ///
```

Comment:
I could add more details:

```suggestion
    /// TypeScript variable declaration without any type annotation and initialization have the `any` type.
    /// The any type in TypeScript is a dangerous “escape hatch” from the type system.
    /// Using any disables many type checking rules and is generally best used only as a last resort or when prototyping code.
    /// TypeScript’s `--noImplicitAny` compiler option doesn't report this case.
```